pkg/collector/corechecks/oracle/activity.go

```diff
@@ -0,0 +1,344 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package oracle
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle/common"
+	"github.com/DataDog/datadog-agent/pkg/obfuscate"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ActivitySnapshot is a payload containing database activity samples. It is parsed from the intake payload.
+// easyjson:json
+type ActivitySnapshot struct {
+	Metadata
+	// Tags should be part of the common Metadata struct but because Activity payloads use a string array
+	// and samples use a comma-delimited list of tags in a single string, both flavors have to be handled differently
+	Tags               []string            `json:"ddtags,omitempty"`
+	CollectionInterval float64             `json:"collection_interval,omitempty"`
+	OracleActivityRows []OracleActivityRow `json:"oracle_activity,omitempty"`
+}
+
+const ACTIVITY_QUERY = `SELECT /* DD_ACTIVITY_SAMPLING */
```

Comment:
One other thing, that we should emit from this code is a new `gauge` metric `oracle.connections` && `oracle.max_connections`

1. `oracle.connections` -> a `gauge` (unit name `connection`) representing the number of **active** connections to this database. If DBM is enabled, this metric is tagged with `status`, `app`, `db` and `user` (for `db` you can use the `pdb` name) 

2. `oracle.max_connections` -> a `gauge` representing the maximum number of **client connections** allowed to this database. 

we do this for postgres, sqlserver and mysql and serveral front end UI components rely on these metrics being present. You can use the output from this query to emit the metric when the check runs

Reply:
created https://datadoghq.atlassian.net/browse/DBM-2306 for that

Reply:
:awesome: